src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -3154,17 +3156,42 @@ static Block.Builder lowerBindingPattern(Block.Builder endNoMatchBlock, Block.Bu
                 Block.Builder nextBlock = currentBlock.block();
 
                 // Check if instance of target type
-                currentBlock.op(conditionalBranch(currentBlock.op(CoreOp.instanceOf(targetType, target)),
-                        nextBlock.successor(), endNoMatchBlock.successor()));
+                Result p;
+                boolean patternWithPrimitive = false;
+                if (targetType instanceof PrimitiveType tt && target.type() instanceof PrimitiveType st) {
+                    patternWithPrimitive = true;
+                    try {
+                        String s = capitalize(st.toString());
+                        String t = capitalize(tt.toString());
+                        String mn = "is%sTo%sExact".formatted(s, t);
+                        MethodRef mref = MethodRef.method(ExactConversionsSupport.class, mn, boolean.class,
+                                st.toNominalDescriptor().resolveConstantDesc(MethodHandles.lookup()));
```

Comment:
Lets avoid resolution of the primitive type, create the method reference from the type elements e.g.,
```
static final JavaType EXACT_CONVERSIONS_SUPPORT = JavaType.type(ExactConversionsSupport.class);
...
MethodRef mref = MethodRef.method(EXACT_CONVERSIONS_SUPPORT, mn, JavaType.BOOLEAN, st);
```